pkg/apply/apply.go

```diff
@@ -121,3 +133,26 @@ func EnvVars(e ...corev1.EnvVar) []corev1ac.EnvVarApplyConfiguration {
 	}
 	return envs
 }
+
+func FromTypedObjWithFields[A any](typed client.Object, fields ...string) (*A, error) {
+	u, err := runtime.DefaultUnstructuredConverter.ToUnstructured(typed)
+	if err != nil {
+		return nil, err
+	}
+	templateSpec, ok, err := unstructured.NestedFieldCopy(u, fields...)
```

Comment:
Maybe change to variable name so it's generic?

Reply:
Oh, good catch

Reply:
Done.